src/components/Molecules/Countdown/Countdown.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Text from '../../Atoms/Text/Text';
+import { Wrapper, Digits } from './Countdown.style';
+
+const Countdown = ({
+  endDate, color, endMessage, introMessage
+}) => {
+  const [countdownEnd, setCountdownEnd] = useState(false);
+  const [countdownTime, setCountdownTime] = useState({});
+
+  const countDownDate = new Date(endDate).getTime();
+  // Keep number with two digits
+  const formatNumber = number => (`0${number}`).slice(-2);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const now = new Date().getTime();
+      const distance = countDownDate - now;
+
+      const days = formatNumber(Math.floor(distance / (1000 * 60 * 60 * 24)));
+      const hours = formatNumber(Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)));
+      const minutes = formatNumber(Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60)));
+      const seconds = formatNumber(Math.floor((distance % (1000 * 60)) / 1000));
```

Comment:
Is it worth unit testing this stuff or using a library e.g. [date-fns](https://date-fns.org/docs/)?

Reply:
I thought about using `momentjs` (we use it in cr.com) but I didn't want to add another dependency.. Do you think is a risk leaving as it is?

Reply:
I think it should be tested tbh. So that we can be confident it works, and confirm that it works as expected across time zones and with daylight saving changes.

You could extract the calculation part into a function outside of the component and export that to test it.

& I think using a library could make it a lot easier to follow and maintain. I'm not sure whether another dependency is much of an issue? `date-fns` and `moment` are both included in `yarn.lock` so they are already dependencies of depedencies

Reply:
TBF it looks like they are dependencies of cypress, so I guess they aren't being used by the components. But to my mind, the benefits of using them outweigh any costs, although I could definitely be wrong...

Reply:
I will try and use `moment`.

Reply:
Cool. The reason I mentioned date-fns btw is that it's more lightweight, you can just import the individual functions you need, so I _think_ it would lead to a smaller package